pkgs/applications/networking/browsers/firefox/wrapper.nix

```diff
@@ -83,6 +113,245 @@ let
+                installation_mode = "blocked";
+              };
+
+          } // lib.foldr (e: ret:
+              ret // {
+                "${e.extid}" = {
+                  installation_mode = "allowed";
+                };
+              }
+            ) {} extraExtensions;
+          }
+      ) // lib.optionalAttrs noNewProfileOnFFUpdate (
+        {
+          LegacyProfiles = true;
+        }
+      ) // lib.optionalAttrs disableFirefoxUpdatePage (
+        {
+          OverridePostUpdatePage = "";
+        }
+      ) // lib.optionalAttrs disablePocket (
+        {
+          DisablePocket = true;
+        }
+      ) // lib.optionalAttrs disableTelemetry (
+        {
+          DisableTelemetry = true;
+        }
+      ) // lib.optionalAttrs disableFirefoxStudies (
+        {
+          DisableFirefoxStudies = true;
+        }
+      ) // lib.optionalAttrs disableFirefoxSync (
+        {
+          DisableFirefoxAccounts = true;
+        }
+      ) // lib.optionalAttrs useSystemCertificates (
+        {
+          # Disable useless firefox certificate store
+          Certificates = {
+            ImportEnterpriseRoots = true;
+          };
+        }
+      ) // lib.optionalAttrs (
+        if lib.count (x: true) extraExtensions > 0 then true else false) (
+        {
+          # Don't try to update nix installed addons
+          DisableSystemAddonUpdate = true;
+
+          # But update manually installed addons
+          ExtensionUpdate = false;
+        }
+      ) // lib.optionalAttrs dontCheckDefaultBrowser (
+        {
+          DontCheckDefaultBrowser = true;
+        }
+      )// lib.optionalAttrs disableDNSOverHTTPS (
+        {
+          DNSOverHTTPS = {
+            Enabled = false;
+          };
+        }
+      ) // lib.optionalAttrs clearDataOnShutdown (
+        {
+          SanitizeOnShutdown = true;
+        }
+      ) // lib.optionalAttrs disableFeedbackCommands (
+        {
+          DisableFeedbackCommands = true;
+        }
+      ) // lib.optionalAttrs ( if homepage == "" then false else true) (
+        {
+          Homepage = {
+            URL = homepage;
+            Locked = true;
+          };
+        }
+      ) // extraPolicies ;} ;
+
+
+      extensions = builtins.map (a:
+        if ! (builtins.hasAttr "signed" a) || ! (builtins.isBool a.signed) then
+          throw "Addon ${a.pname} needs boolean attribute 'signed' "
+        else if ! (builtins.hasAttr "extid" a) || ! (builtins.isString a.extid) then
+          throw "Addon ${a.pname} needs a string attribute 'extid'"
+        else if a.signed == false && !allowNonSigned then
+          throw "Disable signature checking in firefox if you want ${a.pname} addon"
+        else  a
+      ) extraExtensions;
+
+      policiesJson = builtins.toFile "policies.json"
+        (builtins.toJSON enterprisePolicies);
+
+      mozillaCfg = builtins.toFile "mozilla.cfg" ''
+// First line must be a comment
```

Comment:
I think `''` strips indentation if it is equal throught the string, so no need to have zero indentation here